src/nodes/NodeConnectionManager.ts

```diff
@@ -700,19 +741,19 @@ class NodeConnectionManager {
    * @param ctx
    * @returns ConnectionAndLock that was created or exists in the connection map
    */
-  protected getConnectionWithAddress(
+  protected getConnectionWithAddresses(
     targetNodeId: NodeId,
-    address: NodeAddress,
+    addresses: Array<NodeAddress>,
     ctx?: Partial<ContextTimed>,
   ): PromiseCancellable<ConnectionAndTimer>;
   @timedCancellable(
     true,
     (nodeConnectionManager: NodeConnectionManager) =>
       nodeConnectionManager.connectionConnectTimeoutTime,
   )
-  protected async getConnectionWithAddress(
+  protected async getConnectionWithAddresses(
     targetNodeId: NodeId,
-    address: NodeAddress,
+    addresses: Array<NodeAddress>,
     @context ctx: ContextTimed,
   ): Promise<ConnectionAndTimer> {
     const targetNodeIdString = targetNodeId.toString() as NodeIdString;
```

Comment:
Here too, empty array shouldn't be allowed so check and throw.

Reply:
```ts
if (addresses.length === 0) {
      throw new nodesErrors.ErrorNodeConnectionManagerNodeAddressRequired();
    }
```